sys/log/full/src/log.c

```diff
@@ -692,7 +679,9 @@ log_append_prepare(struct log *log, uint8_t module, uint8_t level,
 #endif
 
 #if MYNEWT_VAL(LOG_FLAGS_TRAILER_SUPPORT)
-    ue->ue_flags |= LOG_FLAGS_TRAILER_SUPPORT;
+    if (log->l_th) {
+        ue->ue_flags |= LOG_FLAGS_TRAILER_SUPPORT;
```

Comment:
This has the same issue as previous code. If application registers callbacks, the flag will be set for each new log entry, even if the application doesn't append any trailer data via the callback.

Reply:
If the application has registered trailer callbacks for that particular log, that means the application knows how to handle the trailer data. I really don't get your point. Please provide an alternate approach.